### DISCUSSION TITLE
Running codeactions on save

### DISCUSSION THREAD

**Discussion by abenz1267** [mfussenegger/nvim-jdtls#285] [General]
Hi,

how would you achieve running specific codeactions on save? F.e. adding final modifiers or removing unused imports. I figured you can just call `vim.lsp.buf.code_action({filter = finalModifiers, apply = true})` where `finalModifiers` is a function that picks the correct code-actions. But f.e. you can't really do `vim.lsp.buf.format()` afterwards, it'll break the code. I highly assume because the 2 actions are interfering with another.

How would you chain running those codeactions on save?

Regards

**Comment by mfussenegger** [comment]
I don't think there's currently a good way to do that because code-actions can require multiple request/response roundtrips. It would probably require some `after_apply` handler for `code_action` in neovim core.

**Comment by abenz1267** [comment]
Yeah, i was afraid of something like that. Although it doesn't seem like a super special use-case....
